WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,145 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.example.android.wearable.alpha.editor
 
 import android.os.Bundle
 import android.util.Log
+import android.view.View
 import androidx.activity.ComponentActivity
-import com.example.android.wearable.alpha.R
+import androidx.lifecycle.lifecycleScope
+import com.example.android.wearable.alpha.data.watchface.ColorStyleIdAndResourceIds
+import com.example.android.wearable.alpha.databinding.ActivityWatchFaceConfigBinding
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+import com.example.android.wearable.alpha.editor.WatchFaceConfigViewModel.Companion.MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
-// TODO (codingjeremy): Add activity support to edit watch face (follow up PR).
+/**
+ * Allows user to edit certain parts of the watch face (color style, ticks displayed, minute arm
+ * length) by using the [WatchFaceConfigViewModel]. (All widgets are disabled until data is loaded.)
+ */
 class WatchFaceConfigActivity : ComponentActivity() {
+    private lateinit var viewModel: WatchFaceConfigViewModel
+    private lateinit var binding: ActivityWatchFaceConfigBinding
+
+    private var initialDataLoaded = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_watch_face_config)
         Log.d(TAG, "onCreate()")
+
+        binding = ActivityWatchFaceConfigBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Disable widgets until data loads and values are set.
+        binding.colorStylePickerButton.isEnabled = false
+        binding.ticksEnabledSwitch.isEnabled = false
+        binding.minuteHandLengthSlider.isEnabled = false
+
+        // Set max and min.
+        binding.minuteHandLengthSlider.valueTo = MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.valueFrom = MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.value = MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+
+        binding.minuteHandLengthSlider.addOnChangeListener { slider, value, fromUser ->
+            Log.d(TAG, "addOnChangeListener(): $slider, $value, $fromUser")
+            viewModel.setMinuteHandArmLength(value)
+        }
+
+        lifecycleScope.launch(Dispatchers.Main.immediate) {
+            viewModel = WatchFaceConfigViewModel(
+                    activity = this@WatchFaceConfigActivity,
+                    editIntent = intent
+                )
```

Comment:
We should never be constructing a AAC `ViewModel` directly, nor should we be giving it a reference to an `Activity` or `View`.

Reply:
Whoops, good catch, switched to property delegation.

I'm a little stuck on the other side. The EditorSession (basically the DB) needs an instance of ComponentActivity and Intent... I have a bug for it.